cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1642,6 +1642,83 @@ const char* convert_toUTF8(int64_t context, const char* value, int32_t value_len
   return value;
 }
 
+// Calculate the levenshtein distance between two string values
+FORCE_INLINE
+gdv_int32 levenshtein_utf8_utf8(int64_t context, const char* in1, int32_t in1_len,
+                                const char* in2, int32_t in2_len) {
+  if (in1_len < 0 || in2_len < 0) {
+    gdv_fn_context_set_error_msg(context, "String length must be greater than 0");
+    return 0;
+  }
+
+  // Check input size 0
+  if (in1_len == 0) {
+    return in2_len;
+  }
+  if (in2_len == 0) {
+    return in1_len;
+  }
+
+  int* ptr = new int[(in2_len + 1) * 2];
```

Comment:
There is one more optimization possible (looking at the java code). what if in1_len is far lesser than in2_len? Maybe doing an initial swap and using the bigger array in the outer loop and smaller array in the inner loop may make it more efficient. Or are you planning to make sure in2 is always than in1 when calling the method?

Reply:
Okay, it makes sense! I did the exchange using a boolean variable and two integers so I don't need to copy anything (besides the input sizes)
See if it looks good, please!